### PR TITLE
add acorn run --update/-u flag to update app if exists

### PR DESCRIPTION
**Pull request by iwilltry42**
- triggers acorn update logic if app exists
- does not work with --dev/-i
- requires --name/-n to be specified

Ref #466 

## Notes

- This is not 100% like `kubectl apply`, as that one does a 3-way patch merge, where it takes into account the `last-applied-configuration` annotation. This PR does not consider such annotations.

### CODE REVIEW

**Review by cjellick — COMMENTED** [review]
At first blush, this looks good to me. I'm going to get @ibuildthecloud's input on it

**Comment by ibuildthecloud** [comment]
I really want this, but the approach should be different. "Run -u" and "update" should be slightly different. Basically "run -u" should be like k8s update and update is like k8s patch. 

update today will only change the fields you specify and not touch the other fields. Run -u should ensure all fields exactly match the arguments. So that means run -u will reset fields back to default values if the user didn't specify the flag.

**Comment by iwilltry42** [comment]
@ibuildthecloud does this come close to what you'd expect?

File: pkg/client/client.go [hunk 1]
```diff
@@ -101,6 +101,7 @@ type AppUpdateOptions struct {
 	DevMode         *bool
 	Image           string
 	TargetNamespace string
+	Reset           bool // Reset is used to indicate whether the update should be a patch (reset=false: only change specified fields) or a full update (reset=true: reset unspecified fields to defaults)
```
Comment (ibuildthecloud) [inline pkg/client/client.go:104]: Can you change this to `replace`

**Comment by cjellick** [comment]
@ibuildthecloud mentioned the approach looks right but he wants a new command a different verb that is the equivalent of `update --patch=false`. We decided `acorn replace` should be that verb.

Can you add this @iwilltry42 ?


Moving this out to v0.4.1, which will basically be a milestone for this and one or two other issues that aren't quite ready.

File: docs/docs/100-reference/01-command-line/acorn_update.md [hunk 2]
```diff
@@ -25,6 +25,7 @@ acorn update [flags] APP_NAME [deploy flags]
       --profile strings           Profile to assign default values
   -p, --publish strings           Publish port of application (format [public:]private) (ex 81:80)
   -P, --publish-all               Publish all (true) or none (false) of the defined ports of application
+      --reset
```
[the remark this answers is not in the document]
Comment (iwilltry42) [reply docs/docs/100-reference/01-command-line/acorn_update.md:28]: Left it empty like the other update-only command. Changed now :+1:

File: pkg/client/client.go (hunk 1 above)
Comment (iwilltry42) [reply pkg/client/client.go:104]: Done

**Pull request merged**